viam-orb-slam3.go

```diff
@@ -83,17 +82,14 @@ func init() {
 			return New(ctx, deps, c, logger, false, DefaultExecutableName)
 		},
 	})
-	config.RegisterServiceAttributeMapConverter(slam.Subtype, Model, func(attributes config.AttributeMap) (interface{}, error) {
-		var attrs slamConfig.AttrConfig
-		decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{TagName: "json", Result: &attrs})
-		if err != nil {
-			return nil, err
-		}
-		if err := decoder.Decode(attributes); err != nil {
-			return nil, err
-		}
-		return &attrs, nil
-	}, &slamConfig.AttrConfig{})
+	config.RegisterServiceAttributeMapConverter(
```

Comment:
What does `registry.RegisterService` do & why is this code changing?

Reply:
This code is changing as part of https://viam.atlassian.net/browse/RSDK-2155. Prior to this, we had been using a hack where we use json serialization/deserialization as a way to convert the passed attributes to our attribute format. With the attribute map converter, we can use more rdk features that include better error handling. This is not strictly necessary but if our module is to be an example, we should be using this.